test/server/http/admin_test.cc

```diff
@@ -931,6 +931,139 @@ TEST_P(AdminInstanceTest, ConfigDumpMaintainsOrder) {
   }
 }
 
+TEST_P(AdminInstanceTest, ConfigDumpFiltersByResource) {
```

Comment:
A 1-2 line `//` comment above each `TEST_P` summarizing what is validated would be helpful.